src/arrayMethodSort.js

```diff
@@ -4,8 +4,31 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => {
+      if (String(a) < String(b)) {
+        return -1;
+      } else if (String(a) > String(b)) {
+        return 1;
+      }
+
+      return 0;
+    },
+  ) {
+    for (let i = 0; i < this.length - 1; ) {
+      const a = this[i];
+      const b = this[i + 1];
+
+      if (compareFunction(a, b) > 0) {
+        this[i] = b;
+        this[i + 1] = a;
+        i = 0;
+      } else {
+        i++;
+      }
+    }
```

Comment:
The current sorting algorithm is a bubble sort, which is not efficient for larger arrays. While it will work for small arrays, it is not the most performant for larger ones. Consider implementing a more efficient sorting algorithm like quicksort or mergesort for better performance on larger datasets.